### PR TITLE
Netsuite: sort form-types values based on the type definition as the XML should be sent sorted

### PR DESCRIPTION
**Pull request by omrilit**

### CODE REVIEW

File: packages/netsuite-adapter/src/transformer.ts [hunk 1]
```diff
@@ -137,6 +178,10 @@ export const toCustomizationInfo = (instance: InstanceElement): CustomizationInf
     transformFunc: transformPrimitive,
   }) || {}
 
-  const values = restoreAttributes(transformedValues, instance.type, instance.elemID)
-  return { typeName: instance.type.elemID.name, values }
+  const typeName = instance.type.elemID.name
+
+  const values = shouldSortValues(typeName)
+    ? sortValuesBasedOnType(typeName, transformedValues)
```
Comment (ori-moisis) [inline packages/netsuite-adapter/src/transformer.ts:184]: why use the type name and not the type from `instance.type`?
That way we won't have to create a dummy type as well

File: packages/netsuite-adapter/src/transformer.ts [hunk 2]
```diff
@@ -116,6 +123,40 @@ export const restoreAttributes = (values: Values, type: ObjectType, instancePath
...
+  }
+
+  const fieldType = field.type
+  if (isListType(fieldType) && isObjectType(fieldType.innerType)) {
+    return makeArray(value).map(val => sortObjectValues(fieldType.innerType as ObjectType, val))
+  }
+  if (isObjectType(fieldType) && _.isPlainObject(value)) {
+    return sortObjectValues(fieldType, value)
+  }
+  return value
```
Comment (ori-moisis) [inline packages/netsuite-adapter/src/transformer.ts:141]: Shouldn't `applyRecursive` take care of the recursion part? why does this function have to implement the recursive logic internally as well?

File: packages/netsuite-adapter/src/transformer.ts [hunk 3]
```diff
@@ -116,6 +121,32 @@ export const restoreAttributes = (values: Values, type: ObjectType, instancePath
...
+      return _.fromPairs(fieldsOrder
+        .map(fieldName => [fieldName, value[fieldName]])
+        .filter(([_fieldName, val]) => !_.isUndefined(val)))
+    }
+    return value
+  }
+
+  return transformValues(
+    { type: topLevelType, values, transformFunc: sortValues, pathID: instancePath }
+  ) ?? {}
```
Comment (ori-moisis) [inline packages/netsuite-adapter/src/transformer.ts:142]: using the fields as the basis for order means it possibly removed values.
this is OK if we want this behavior, but maybe to be consistent about this intention we should call `transformValues` with `strict = true`
wdyt?

Comment (omrilit) [reply packages/netsuite-adapter/src/transformer.ts]: Done

**Pull request merged**